superset/datasets/schemas.py

```diff
@@ -204,8 +204,13 @@ def fix_extra(self, data: dict[str, Any], **kwargs: Any) -> dict[str, Any]:
         """
         Fix for extra initially being exported as a string.
         """
-        if isinstance(data.get("extra"), str):
-            data["extra"] = json.loads(data["extra"])
+        if "extra" in data:
+            extra = data["extra"]
+            if isinstance(extra, str):
+                if extra.strip():
+                    data["extra"] = json.loads(extra)
+                else:
+                    data["extra"] = {}
```

Comment:
You can use the walrus operator `:=` here:

```suggestion
        if extra := data.get("extra"):
            if isinstance(extra, str):
                data["extra"] = json.loads(extra) if extra.strip() else {}
```

Reply:
Thanks! I tested this locally and ended up using `try/except` because if `data["extra"] = ''` the `if` block wouldn't be executed.